ptrace/debugger/child.py

```diff
@@ -87,6 +88,27 @@ def _createParent(pid, errpipe_read):
         raise child_exception
 
 
+def _closeFdsExcept(fds, ignore_fds):
+    for fd in fds:
+        if fd not in ignore_fds:
+            try:
+                close(fd)
+            except OSError:
+                pass
+
+
+def _closeFds(ignore_fds):
+    for path in ['/proc/self/fd', '/dev/fd']:
```

Comment:
/dev/fd is symlink in Linux. 
Other systems like BSD and Mac have /dev/fd.
On FreeBSD until fdescfs is mounted it contains only static descriptors 0, 1, 2 so there will be needed another test, otherwise it wont close any descriptor.
I will look at python subprocess.